On-premise installation failed at reference-visualization : Building Angular client-side code

One step installation is not happeining Failed at 
TASK [reference-visualization : Building Angular client-side code] *************
fatal: [localhost]: FAILED! => {"changed": true, "cmd": ["ng", "build"], "delta": "0:02:24.506518", "end": "2022-09-27 13:00:03.022463", "msg": "non-zero return code", "rc": 1, "start": "2022-09-27 12:57:38.515945", "stderr": "- Generating browser application bundles (phase: setup)...\nProcessing legacy \"View Engine\" libraries:\n- @ng-bootstrap/ng-bootstrap [es2015/esm2015] (git+https://github.com/ng-bootstrap/ng-bootstrap.git)\nEncourage the library authors to publish an Ivy distribution.\n\n- ng-circle-progress [es2015/esm2015] (git+https://github.com/bootsoon/ng-circle-progress.git)\n\n- ng2-charts [es2015/esm2015] (git+https://github.com/valor-software/ng2-charts.git)\n\n- ng-multiselect-dropdown [es2015/esm2015] (git+https://github.com/nileshpatel17/ng-multiselect-dropdown.git)\n\n- ngx-daterangepicker-material [es2015/esm2015] (git+https://github.com/fetrarij/ngx-daterangepicker-material.git)

 "Error: src/environments/environment.prod.ts:19:3 - error TS1117: An object literal cannot have multiple properties with the same name.", "", "19   apiEndpoint: \"http://localhost:3004/core_api\",", "     ~~~~~~~~~~~"], "stdout": "", "stdout_lines": []}

--- a/devops/ansible/roles/aggregated_data_processor/tasks/nifi_aggregate.yml
+++ b/devops/ansible/roles/aggregated_data_processor/tasks/nifi_aggregate.yml
@@ -1,4 +1,4 @@
 ---
 - name: Run aggregated data processor python code
-  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utililty.py aggregated_data_processor_group static_data_parameters  cQube_data_storage 0 {{ storage_type }}"
+  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utility.py aggregated_data_processor_group static_data_parameters  cQube_data_storage 0 {{ storage_type }}"
   tags: [ install, update ]

--- a/devops/ansible/roles/api/tasks/configure_core_api.yml
+++ b/devops/ansible/roles/api/tasks/configure_core_api.yml
@@ -30,7 +30,7 @@
   blockinfile:
     path: "{{base_dir}}/cqube/apis/core-api/.env"
     block: |
-      PORT = 3001
+      PORT = 3004
       ACCESS_KEY_ID = "{{ s3_access_key }}"
       SECRET_ACCESS_KEY  = "{{ s3_secret_key }}"
       OUTPUT_BUCKET = "{{ s3_output_bucket }}"

--- a/devops/ansible/roles/api/tasks/configure_node_api.yml
+++ b/devops/ansible/roles/api/tasks/configure_node_api.yml
@@ -36,7 +36,7 @@
   blockinfile:
     path: "{{base_dir}}/cqube/apis/node-api/.env"
     block: |
-      STORAGE_SERVICE = {% if storage_type == "azure" %}AZURE_DATA_LAKE{% endif %}{% if storage_type == "s3" %}AWS_S3{% endif %}
+      STORAGE_SERVICE = "{% if storage_type == 'azure' %}AZURE_DATA_LAKE{% endif %}{% if storage_type == 's3' %}AWS_S3{% endif %}"
       AZURE_STORAGE_CONNECTION_STRING = "{{ azure_storage_connection_string }}"
       OUTPUT_CONTAINER = "{{ azure_output_container }}"
       INPUT_CONTAINER = "{{ azure_input_container }}"

--- a/devops/ansible/roles/attendance/tasks/nifi_attendance.yml
+++ b/devops/ansible/roles/attendance/tasks/nifi_attendance.yml
@@ -1,5 +1,5 @@
 ---
 - name: Run student attendance python code
-  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utililty.py student_attendance_transformer student_attendance_parameters cQube_data_storage 4557 {{ storage_type }}"
+  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utility.py student_attendance_transformer student_attendance_parameters cQube_data_storage 4557 {{ storage_type }}"
   tags: [ install, update ]
 

--- a/devops/ansible/roles/composite/tasks/nifi_composite.yml
+++ b/devops/ansible/roles/composite/tasks/nifi_composite.yml
@@ -1,4 +1,4 @@
 ---
 - name: Run composite python code
-  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utililty.py composite_transformer composite_parameters cQube_data_storage 0"
+  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utility.py composite_transformer composite_parameters cQube_data_storage 0"
   tags: [ install, update ]

--- a/devops/ansible/roles/configurable/tasks/nifi_configurable.yml
+++ b/devops/ansible/roles/configurable/tasks/nifi_configurable.yml
@@ -1,4 +1,4 @@
 ---
 - name: Run configurable python code
-  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utililty.py configurable_datasource static_data_parameters cQube_data_storage 0 {{ storage_type }}"
+  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utility.py configurable_datasource static_data_parameters cQube_data_storage 0 {{ storage_type }}"
   tags: [ install, update ]

--- a/devops/ansible/roles/cqube_data/tasks/nifi_cqube_data.yml
+++ b/devops/ansible/roles/cqube_data/tasks/nifi_cqube_data.yml
@@ -1,3 +1,3 @@
 - name: Run cqube data storage python code
-  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utililty.py cQube_data_storage cQube_data_storage_parameters 4559"
+  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utility.py cQube_data_storage cQube_data_storage_parameters 4559"
   tags: [ install, update ]

--- a/devops/ansible/roles/crc/tasks/nifi_crc.yml
+++ b/devops/ansible/roles/crc/tasks/nifi_crc.yml
@@ -1,5 +1,5 @@
 ---
 - name: Run crc python code
-  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utililty.py crc_transformer crc_parameters cQube_data_storage 4660 {{ storage_type }}"
+  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utility.py crc_transformer crc_parameters cQube_data_storage 4660 {{ storage_type }}"
   tags: [ install, update ]
 

--- a/devops/ansible/roles/data_replay/tasks/nifi_data_replay.yml
+++ b/devops/ansible/roles/data_replay/tasks/nifi_data_replay.yml
@@ -1,4 +1,4 @@
 ---
 - name: Run data replay python code
-  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utililty.py data_replay_transformer data_replay_parameters cQube_data_storage 0 {{ storage_type }}"
+  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utility.py data_replay_transformer data_replay_parameters cQube_data_storage 0 {{ storage_type }}"
   tags: [ install, update ]

--- a/devops/ansible/roles/diksha/tasks/nifi_diksha.yml
+++ b/devops/ansible/roles/diksha/tasks/nifi_diksha.yml
@@ -1,15 +1,15 @@
 - name: Run diksha python code
-  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utililty.py diksha_transformer diksha_parameters cQube_data_storage 4662 {{ storage_type }}"
+  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utility.py diksha_transformer diksha_parameters cQube_data_storage 4662 {{ storage_type }}"
   tags: [ install, update ]
 
 - name: Enable diksha ETB API
-  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utililty.py diksha_transformer {{ storage_type }} ETB API"                                         
+  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utility.py diksha_transformer {{ storage_type }} ETB API"                                         
   tags: [ install, update ]
 
 - name: Enable diksha TPD API
-  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utililty.py diksha_transformer {{ storage_type }} TPD API"                                         
+  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utility.py diksha_transformer {{ storage_type }} TPD API"                                         
   tags: [ install, update ]
 
 - name: Run diksha custom python code
-  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utililty.py diksha_transformer_custom diksha_parameters cQube_data_storage 0"
+  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utility.py diksha_transformer_custom diksha_parameters cQube_data_storage 0"
   tags: [ install, update ]

--- a/devops/ansible/roles/dummy_connections/tasks/nifi_dummy_connections.yml
+++ b/devops/ansible/roles/dummy_connections/tasks/nifi_dummy_connections.yml
@@ -1,8 +1,8 @@
 - name: run python code to create dummy connections
-  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utililty.py cQube_data_storage"
+  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utility.py cQube_data_storage"
   tags: [ install, update ]
 
 - name: run python code to start telemetry
-  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_start_pg.py cqube_telemetry_transformer RUNNING"
+  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utility.py cqube_telemetry_transformer RUNNING"
   when: telemetry
   tags: [ install, update ]

--- a/devops/ansible/roles/infra/tasks/nifi_infra.yml
+++ b/devops/ansible/roles/infra/tasks/nifi_infra.yml
@@ -1,4 +1,4 @@
 ---
 - name: Run infra python code
-  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utililty.py infra_transformer infra_parameters cQube_data_storage 4661 {{ storage_type }}"
+  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utility.py infra_transformer infra_parameters cQube_data_storage 4661 {{ storage_type }}"
   tags: [ install, update ]

--- a/devops/ansible/roles/nifi_scheduler/tasks/nifi_scheduler.yml
+++ b/devops/ansible/roles/nifi_scheduler/tasks/nifi_scheduler.yml
@@ -1,4 +1,3 @@
 - name: Run python code for nifi scheduler
   shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_scheduler.py"
   tags: [ install, update ]
-

--- a/devops/ansible/roles/pat/tasks/nifi_pat.yml
+++ b/devops/ansible/roles/pat/tasks/nifi_pat.yml
@@ -1,5 +1,4 @@
 ---
 - name: Run pat python code
-  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utililty.py pat_transformer pat_parameters cQube_data_storage 4665 {{ storage_type }}"
+  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utility.py pat_transformer pat_parameters cQube_data_storage 4665 {{ storage_type }}"
   tags: [ install, update ]
-

--- a/devops/ansible/roles/progresscard/tasks/nifi_progresscard.yml
+++ b/devops/ansible/roles/progresscard/tasks/nifi_progresscard.yml
@@ -1,4 +1,4 @@
 ---
 - name: Run progress card python code
-  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utililty.py progress_card_transformer progress_card_parameters cQube_data_storage 0"
+  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utility.py progress_card_transformer progress_card_parameters cQube_data_storage 0"
   tags: [ install, update ]

--- a/devops/ansible/roles/reference-visualization/templates/dashboard_environment.prod_local.ts.j2
+++ b/devops/ansible/roles/reference-visualization/templates/dashboard_environment.prod_local.ts.j2
@@ -1,8 +1,9 @@
 export const environment = {
   production: true,
-  apiEndpoint: "{{ protocol }}://{{ api_endpoint }}:3000/api",
+  apiURL: "{{ protocol }}://{{ api_endpoint }}:3000/api",
   appUrl: "{{ protocol }}://{{ api_endpoint }}:4200",
-  adminUrl: "http://{{ proxy_host }}:4201",
+  loginUrl: "{{ protocol }}://{{ api_endpoint }}:4001/login-api",
+  adminUrl: "{{ protocol }}://{{ proxy_host }}:4201",
   keycloakUrl: "{{ protocol }}://{{ api_endpoint }}:8080/auth",
   realm: "{{ realm_name }}",
   config: "{{ access_type }}",

--- a/devops/ansible/roles/reference-visualization/templates/dashboard_environment_local.ts.j2
+++ b/devops/ansible/roles/reference-visualization/templates/dashboard_environment_local.ts.j2
@@ -1,8 +1,9 @@
 export const environment = {
   production: false,
-  apiEndpoint: "{{ protocol }}://{{ api_endpoint }}:3000/api",
+  apiURL: "{{ protocol }}://{{ api_endpoint }}:3000/api",
   appUrl: "{{ protocol }}://{{ api_endpoint }}:4200",
-  adminUrl: "http://{{ proxy_host }}:4201",
+  adminUrl: "{{ protocol }}://{{ proxy_host }}:4201",
+  loginUrl: "{{ protocol }}://{{ api_endpoint }}:4001/login-api",
   keycloakUrl: "{{ protocol }}://{{ api_endpoint }}:8080/auth",
   realm: "{{ realm_name }}",
   config: "{{ access_type }}",

--- a/devops/ansible/roles/sat/tasks/nifi_sat.yml
+++ b/devops/ansible/roles/sat/tasks/nifi_sat.yml
@@ -1,4 +1,4 @@
 ---
 - name: Run sat python code
-  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utililty.py sat_transformer sat_parameters cQube_data_storage 4668 {{ storage_type }}"
+  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utility.py sat_transformer sat_parameters cQube_data_storage 4668 {{ storage_type }}"
   tags: [ install, update ]

--- a/devops/ansible/roles/static/tasks/nifi_static.yml
+++ b/devops/ansible/roles/static/tasks/nifi_static.yml
@@ -1,4 +1,4 @@
 ---
 - name: Run static python code
-  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utililty.py static_data_transformer static_data_parameters  cQube_data_storage 0 {{ storage_type }}"
+  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utility.py static_data_transformer static_data_parameters  cQube_data_storage 0 {{ storage_type }}"
   tags: [ install, update ]

--- a/devops/ansible/roles/teacher_attendance/tasks/nifi_teacher_attendance.yml
+++ b/devops/ansible/roles/teacher_attendance/tasks/nifi_teacher_attendance.yml
@@ -1,5 +1,5 @@
 ---
 - name: Run teacher attendance python code
-  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utililty.py teacher_attendance_transformer teacher_attendance_parameters cQube_data_storage 4666 {{ storage_type }}"
+  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utility.py teacher_attendance_transformer teacher_attendance_parameters cQube_data_storage 4666 {{ storage_type }}"
   tags: [ install, update ]
 

--- a/devops/ansible/roles/telemetry/tasks/nifi_telemetry.yml
+++ b/devops/ansible/roles/telemetry/tasks/nifi_telemetry.yml
@@ -1,5 +1,5 @@
 ---
 - name: Run telemetry python code
-  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utililty.py cqube_telemetry_transformer cqube_telemetry_parameters cQube_data_storage 0 {{ storage_type }}"
+  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utility.py cqube_telemetry_transformer cqube_telemetry_parameters cQube_data_storage 0 {{ storage_type }}"
   tags: [ install, update ]
 

--- a/devops/ansible/roles/transaction_and_aggregation/tasks/nifi_transaction_and_aggregation.yml
+++ b/devops/ansible/roles/transaction_and_aggregation/tasks/nifi_transaction_and_aggregation.yml
@@ -1,3 +1,3 @@
 - name: Run nifi_transaction_and_aggregation python code
-  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utililty.py transaction_and_aggregation transaction_and_aggregation_parameters cQube_data_storage 4700"
+  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utility.py transaction_and_aggregation transaction_and_aggregation_parameters cQube_data_storage 4700"
   tags: [ install, update ]

--- a/devops/ansible/roles/udise/tasks/nifi_udise.yml
+++ b/devops/ansible/roles/udise/tasks/nifi_udise.yml
@@ -1,5 +1,5 @@
 ---
 - name: Run udise python code
-  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utililty.py udise_transformer udise_parameters cQube_data_storage 4664 {{ storage_type }}"
+  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utility.py udise_transformer udise_parameters cQube_data_storage 4664 {{ storage_type }}"
   tags: [ install, update ]
 

--- a/devops/ansible/roles/validate/tasks/nifi_validate.yml
+++ b/devops/ansible/roles/validate/tasks/nifi_validate.yml
@@ -1,4 +1,4 @@
 ---
 - name: Run validate datasource python code
-  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utililty.py validate_datasource validate_datasource_parameters cQube_data_storage 4669 {{ storage_type }}"
+  shell: "{{ base_dir }}/cqube/emission_app/flaskenv/bin/python {{ base_dir }}/cqube/emission_app/python/nifi_utility.py validate_datasource validate_datasource_parameters cQube_data_storage 4669 {{ storage_type }}"
   tags: [ install, update ]

